src/transaction/lock.rs

```diff
@@ -21,67 +21,77 @@ pub async fn resolve_locks(
     pd_client: Arc<impl PdClient>,
 ) -> Result<bool> {
     let ts = pd_client.clone().get_timestamp().await?;
-    let mut has_live_locks = false;
-    let expired_locks = locks.into_iter().filter(|lock| {
+    let locks_len = locks.len();
+
+    struct RichLock {
+        primary_key: Key,
+        region: Region,
+        lock_version: u64,
+        txn_size: u64,
+    }
+
+    let mut expired_locks = vec![];
+    for lock in locks {
```

Comment:
I think we can iterate only once. For every expired lock, check if it's big enough to use the lite way, store them in a map with `(Region, lock_version)` as the key. The value is only the primary key for the normal way, and primary key plus all the locks in the region for the lite way. Then use different ways to handle them.

First collecting expired and then using `group by` look too heavy for this requirement.